File: Medium/129-sum-root-to-leaf-numbers/try-3.cpp
/*
 * Problem #129: Sum Root to Leaf Numbers
 * Difficulty: Medium
 * Submission: Try 3
 * status: Accepted
 * Language: cpp
 * Date: 24/11/2025, 11:35:51
 * Link: https://leetcode.com/problems/sum-root-to-leaf-numbers/
 */

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int sumNumbers(TreeNode* root) {
        return dfs(root,0);
    }
    int dfs(TreeNode* root,int str){
        if(!root) return 0;
        str = str*10 + root->val;
        if(!root->left && !root->right) return str;
        return dfs(root->left,str) + dfs(root->right,str);
    }
};

File: Medium/114-flatten-binary-tree-to-linked-list/try-3.py
/*
 * Problem #114: Flatten Binary Tree to Linked List
 * Difficulty: Medium
 * Submission: Try 3
 * status: Accepted
 * Language: python3
 * Date: 07/01/2025, 17:54:00
 * Link: https://leetcode.com/problems/flatten-binary-tree-to-linked-list/
 */

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def flatten(self, root: Optional[TreeNode]) -> None:
        """
        Do not return anything, modify root in-place instead.
        """
        if root == None:
            return
        
        self.flatten(root.left)
        self.flatten(root.right)

        temp = root.right
        root.right = root.left
        root.left = None
        current = root

        while current.right:
            current = current.right
        current.right = temp

